dev/bus/tests/serial_unit.cpp

```diff
@@ -0,0 +1,83 @@
+#include <CppUTest/TestHarness.h>
+#include <CppUTest/CommandLineTestRunner.h>
+#include <CppUTestExt/MockSupport.h>
+
+#include "dev/serial.hpp"
+#include "mocks/platform_bus.hpp"
+#include "ecl/thread/semaphore.hpp"
+#include <thread>
+
+// Error code helper
+// TODO: move it to 'utils' headers and protect with check of
+// current test state (enabled or disabled)
+static SimpleString StringFrom(ecl::err err)
+{
+    return SimpleString{ecl::err_to_str(err)};
+}
+
+using serial_t = ecl::serial<platform_mock>;
+
+TEST_GROUP(serial)
+{
+    void setup()
+    {
+        mock().disable();
+        serial_t::init();
+        mock().enable();
+    }
+
+    void teardown()
+    {
+        mock().disable();
+        serial_t::deinit();
+        mock().enable();
+        mock().clear();
+    }
+
+};
+
+TEST(serial, recv_byte)
+{
+    // Internal buffer is empty so the recv_byte() should block
+    // until the platform invokes bus_handler
+    ecl::binary_semaphore recv_status;
+    std::atomic_bool recv_returned;
+    std::thread t([&] {
+            recv_status.wait(); //Wait for the recv_byte() call
+            CHECK_EQUAL(recv_returned, false);
+            platform_mock::invoke(ecl::bus_channel::rx, ecl::bus_event::tc, 1); // Unblock
+        });
+    auto expected = ecl::err::ok;
+    uint8_t byte;
+    recv_status.signal();
+    auto actual = serial_t::recv_byte(byte);
+    recv_returned = true;
+    t.join();
+    CHECK_EQUAL(expected, actual);
```

Comment:
As a note for our offline discussion: let's check that byte received is the same byte that were written by `platform_mock`